packages/excalidraw/components/HintViewer.tsx

```diff
@@ -165,34 +168,39 @@ const getHints = ({
   return null;
 };
 
-export const HintViewer = ({
-  appState,
-  isMobile,
-  device,
-  app,
-}: HintViewerProps) => {
-  const hints = getHints({
-    appState,
-    isMobile,
-    device,
-    app,
-  });
-
-  if (!hints) {
-    return null;
-  }
+export const HintViewer = React.memo(
+  ({ appState, isMobile, device, app }: HintViewerProps) => {
+    const hints = getHints({
+      appState,
+      isMobile,
+      device,
+      app,
+    });
+
+    if (!hints) {
+      return null;
+    }
 
-  const hint = Array.isArray(hints)
-    ? hints
-        .map((hint) => {
-          return getShortcutKey(hint).replace(/\. ?$/, "");
-        })
-        .join(". ")
-    : getShortcutKey(hints);
-
-  return (
-    <div className="HintViewer">
-      <span>{hint}</span>
-    </div>
-  );
-};
+    const hint = Array.isArray(hints)
+      ? hints
+          .map((hint) => {
+            return getShortcutKey(hint).replace(/\. ?$/, "");
+          })
+          .join(". ")
+      : getShortcutKey(hints);
+
+    const hintElement = React.createElement(
+      React.Fragment,
+      {},
+      ...nodesFromTextWithTags(hint, {
+        shortcut: (el) => <span className="HintViewer__key">{el}</span>,
+      }),
+    );
```

Comment:
Why not just do this:

```jsx
const hint_JSX = hint
  .split(/(Space|Alt|Enter|Cmd|Option|Ctrl)/g)
  .map((part, index) => {
    if (index % 2 === 1) {
      return (
        <span key={index} className="HintViewer__key">
          {part}
        </span>
      );
    }
    return part;
  });

return (
  <div className="HintViewer">
    <span>{hint_JSX}</span>
  </div>
);
```

Simpler and we don't need to modify the translation files.

Reply:
ok, there are edge cases I see. Thinking...

Reply:
@dwelle it won't work with single letter shortcuts, or common words that are also keys, like these:
<img width="827" height="111" alt="image" src="https://github.com/user-attachments/assets/88c13866-740a-44ca-802a-3ebba4836cb4" />
<img width="805" height="114" alt="image" src="https://github.com/user-attachments/assets/aed7906d-4480-45aa-aee2-bd8802925078" />